test/e2e/rollout_test.go

```diff
@@ -542,6 +548,231 @@ var _ = Describe("placing wrapped resources using a CRP", Ordered, func() {
+				},
+				{
+					Group:   utils.ClusterRoleGVK.Group,
+					Kind:    utils.ClusterRoleGVK.Kind,
+					Name:    clusterRoleName,
+					Version: utils.ClusterRoleGVK.Version,
+				},
+			}
+		})
+
+		It("create the resources", func() {
+			Expect(hubClient.Create(ctx, &workNamespace)).To(Succeed(), "Failed to create namespace %s", workNamespace.Name)
+			Expect(hubClient.Create(ctx, &clusterRole)).Should(Succeed(), "Failed to create the clusterRole %s", clusterRoleName)
+		})
+
+		It("create the CRP that select the namespace and ClusterRole", func() {
+			crp = buildCRPForSafeRollout()
+			crp.Spec.ResourceSelectors = append(crp.Spec.ResourceSelectors, placementv1beta1.ClusterResourceSelector{
+				Group:   utils.ClusterRoleGVK.Group,
+				Kind:    utils.ClusterRoleGVK.Kind,
+				Name:    clusterRoleName,
+				Version: utils.ClusterRoleGVK.Version,
+			})
+			crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds = ptr.To(60)
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := crpStatusUpdatedActual(wantSelectedResources, allMemberClusterNames, nil, "0")
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("should place the resources on member cluster", func() {
+			name := types.NamespacedName{Name: clusterRoleName}
+			wantClusterRole := &rbacv1.ClusterRole{}
+			Expect(hubClient.Get(ctx, name, wantClusterRole)).Should(Succeed(), "Failed to get the clusterRole on hub cluster")
+
+			for idx := range allMemberClusters {
+				memberCluster := allMemberClusters[idx]
+
+				resourcesPlacedActual := func() error {
+					clusterRole := &rbacv1.ClusterRole{}
+					if err := memberCluster.KubeClient.Get(ctx, name, clusterRole); err != nil {
+						return err
+					}
+					if diff := cmp.Diff(clusterRole, wantClusterRole, ignoreObjectMetaAutoGeneratedFields, ignoreObjectMetaAnnotationField); diff != "" {
+						return fmt.Errorf("clusterRole diff (-got, +want): %s", diff)
+					}
+					return nil
+				}
+				Eventually(resourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on member cluster %s", memberCluster.ClusterName)
+			}
+		})
+
+		It("should update custom resource", func() {
+			Eventually(func() error {
+				cr := &rbacv1.ClusterRole{}
+				err := hubClient.Get(ctx, types.NamespacedName{Name: clusterRoleName}, cr)
+				if err != nil {
+					return err
+				}
+				cr.Rules[0].Verbs[0] = "list" // Previously was "get"
+				return hubClient.Update(ctx, cr)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update ClusterRole")
+		})
+
+		It("should update the resource on member cluster", func() {
+			Consistently(func() error {
+				cr := &rbacv1.ClusterRole{}
+				err := hubClient.Get(ctx, types.NamespacedName{Name: clusterRoleName}, cr)
```

Comment:
I don't get it, why check the resource on hub? Why is the resource supposed to change instantly on the member?

Reply:
I meant to check the resource on member cluster. I was thinking it would be changed pretty quickly since it is one of the resources that be ready and available as soon as they are created.